### ISSUE TITLE
Basic tablet mode support

### ISSUE THREAD

**Issue by mantomas** [regolith-linux/regolith-desktop#396]
I think there should be some basic  support for convertible laptops (touchscreen - tablet mode) in Regolith DE. 
I have Regolith installed from ISO(1.4), but:
- screen auto rotation sometimes freeze and I need terminal command `xrandr -o normal ` to get back to normal
- screen is rotated but touchscreen/touchpad input not, so there is no way to naturally scroll documents or so
- when is display in tablet position, keyboard input is disabled, but touchpad still work
- no onscreen keyboard, even if sett ON in settings

I installed Ubuntu-desktop parallel with Regolith and made test user profile to see the difference. In Ubuntu DE all mentioned works great - rotated input, touchpad off, onscreen keyboard...
So is there any way to get settings/packages from Ubuntu-desktop and make Regolith adopt them? I think everything except onscreen-keyboard should be manageable easily, so any hint where to start?

**Comment by moritzheiber** [comment]
It's highly unlikely this is ever going to get supported any time soon. Regolith relies on `gnome-flashback` for most of its GNOME integration, and as such tablet support isn't a major feature that's on `gnome-flashback`'s roadmap, as far as I know (cc @muktupavels).

Regolith cannot do much about this, and i3 definitely wasn't designed to be used on tablets either.

**Comment by mantomas** [comment]
Ok, I understand @moritzheiber . Maybe my point was, if it's possible to make Regolith configuration better in things, which are already there. Screen orientation sensors are automatically detected, so is touch-screen input, only these two things does not work together.

- screen is rotated automatically out of the box
- touchscreen input isn't rotated, but can be set using "Coordinate Transformation Matrix" in `xinput set-prop`

I don't see under the hood, it's all too complicated for me, but if `gnome-flashback` is build on Gnome 3 as is `ubuntu-desktop` and ubuntu-desktop is working on the same installation fine, can it be configured the same way? 
I really don't mind full touch-control, my point is **hardware support** and **proper input recognition**.

**Comment by moritzheiber** [comment]
I understand your grievances, however, as I said, that's beyond the scope of what Regolith can provide. It relies on other components to provide this functionality, namely gnome-flashback and its integrations. If you think gnome-flashback ought to support certain functionality you'll have to file a feature request with them

**Comment by muktupavels** [comment]
In GNOME 3 these things are managed mostly by `mutter` and/or `gnome-shell`. It might be possible to find related code in mutter/gnome-shell and re-use and/or do something similar in gnome-flashback. Unfortunately I don't have such devices... So it is unlikely that support will get better without ability to test related functionality.

For example touchscreen input problem might be related to this:  
https://gitlab.gnome.org/GNOME/gnome-flashback/-/blob/master/gnome-flashback/libinput-settings/gf-input-settings.c#L1150-1196

When I originally added that code based on mutter GTK did not had source related to tablet. Now there is `GDK_SOURCE_TABLET_PAD` that might be equivalent to `CLUTTER_TABLET_DEVICE` (did not check that)... `CLUTTER_INPUT_DEVICE_MAPPING_ABSOLUTE` seems to be related to `GDesktopTabletMapping`... It might be possible to finish support for tablet.

**Comment by moritzheiber** [comment]
@mantomas Given what @muktupavels has said (they are one of the gnome-flashback developers) would you consider closing this bug in favor of opening it upstream instead? It'll be easier to track its progress then :slightly_smiling_face:

**Comment by mantomas** [comment]
@moritzheiber I'm a bit confused about it all. I partly understand what @muktupavels said, but its above my skill to investigate the problem that way. If you think there is nothing related to Regolith, I don't mind if anybody close this bug and report it where it belong. 
I did some final testing and find that mouse input is recognized correctly on rotated screen. My opinion is that touchscreen is internally recognized the same way as touchpad and because you don't rotate touchpad, the same way as you don't rotate mouse when working on portrait-oriented screen - input device is still in the same position.
I don't need seamless tablet experience, but I think there is just simple need to use any screen with any input device, no mater if i3, KDE, Gnome... Or is it strange to use Gimp on portrait-oriented standalone screen? If there is a link on a web page, does it mater what input device I use to get cursor there? I plugged in my old Wacom tablet and the same result - input is distorted as the touchpad.
If anybody knows, where to fill the issue and what is the correct description of the problem?..
